### DISCUSSION TITLE
[Vue] What's the way to go today !?

### DISCUSSION THREAD

**Discussion by tooppaaa** [storybookjs/storybook#11073] [Help]
Dear Vue community,

Storybook has currently 2 formats for vue when initializing a new project: 

## Vue Single File Component
**When** dependencies are one of vue-loader, vuetify
Uses .vue files with https://vuejs.org/v2/guide/single-file-components.html

## Vue "Classic"
**When** dependencies are one of vue, nuxt
Uses .js files with 
```js
export default {
  name: '',

  data() { return {}; },

  template: ``,

  methods: {},
};
```

As part of the CLI revamp, we're going through these files and wondering if it's worth keeping both ways.
If yes are the dependencies still valid ?

Please upvote the next comments with the format you're using and feel free to provide feedbacks !

Thanks for your help

**Comment by tooppaaa** [comment]
I'm using Vue SFC !

**Comment by tooppaaa** [comment]
I'm not using SFC

**Reply by Aaron-Pool** [reply]
I'm not at all implying that this should be the standard way of doing things, but I've always used jsx exclusively in my Vue storybook pattern library. There's much more flexibility that way, and it actually fits more intuitively into the storybook ecosystem when you really embrace jsx.

That being said, I'm about 97% confident that the _vast_ majority of Vue users use SFCs exclusively, and have never even written a render function. So, while I don't use SFCs personally, I would advise you embrace that as your "typical" case. Additionally, people that use render functions/jsx are probably more familiar with doing advanced setups and will likely be more equipped to figure things out for themselves, without the aid of a "getting started" guide.
